go/cmd/dolt/cli/arg_parser_helpers.go

```diff
@@ -98,6 +98,7 @@ const (
 	RemoteParam      = "remote"
 	BranchParam      = "branch"
 	TrackFlag        = "track"
+	NewFormatFlag    = "new-format"
```

Comment:
(minor) I'm wondering if there's a longer lasting name than "new-format" (and also if it matters). I'm just imagining a year or two out when we may still have this flag around and the format isn't new anymore. I know we'll have made it the default format by then, but just wondering if we should consider something like:

```
dolt init --dolt-format
```

This might be moot if you're already planning on removing that flag after we switch it to the default, but it still might be nice to have a more explicit name. 

Once we flip the default, will there be another option for customers to initialize with the old format? If so... maybe it makes more sense to have something like `--format=ld1` and `--format=dolt`? 

Not a huge deal either way, but worth thinking about the ergonomics and how it'll play out over time.